### PR TITLE
AppVeyorPoc: Navigation done

### PR DESCRIPTION
**Pull request by ILyaIL83**
#First draft of "Web client POC" #44 implementation.

### CODE REVIEW

**Comment by ILyaIL83** [comment]
New commit was made(AppVeyorPoc: Real Rest API used https://github.com/felix-b/NWheels/pull/48/commits/037e0885bef0fb292b7f0f77661898c37f362368), where used real AppVeyor Rest Api.
To run it you have to add Repos\NWheelsFork\Source\Presentation\Web\AppVeyorPoc\src\app\app-veyor-service-token.ts file
`export class AppVeyorServiceToken {
  static token = 'your token';
}`
Token can be taken from here https://ci.appveyor.com/api-token

**Comment by felix-b** [comment]
Get the following error when trying to run ```ng serve```:
```
ERROR in C:/Home/NWheels_ILyaIL83/Source/Presentation/Web/AppVeyorPoc/src/app/app-routing.module.ts (8,32): Cannot find module './projects/build/build.component'.
C:/Home/NWheels_ILyaIL83/Source/Presentation/Web/AppVeyorPoc/src/app/app-routing.module.ts (9,34): Cannot find module './projects/build/console.component'.
C:/Home/NWheels_ILyaIL83/Source/Presentation/Web/AppVeyorPoc/src/app/app-routing.module.ts (10,35): Cannot find module './projects/build/messages.component'.
C:/Home/NWheels_ILyaIL83/Source/Presentation/Web/AppVeyorPoc/src/app/app-routing.module.ts (11,36): Cannot find module './projects/build/artifacts.component'.
```
Full output attached.

[ngserve-error-1.txt](https://github.com/felix-b/NWheels/files/915421/ngserve-error-1.txt)

**Comment by ILyaIL83** [comment]
Fixed.
In root .gitignore we have 
`# Build results

...

build/

...`

So in AppVeyor web application I force all files to be included.
`# Make sure to include everything in src.

!src/**`

**Comment by felix-b** [comment]
Great! I'll try again. On another note, does this POC include testing?

**Comment by ILyaIL83** [comment]
All unit tests, as we discussed, doesn't make sense because this code should be generated.

File: Source/Presentation/Web/AppVeyorPoc/src/app/app-veyor-service-token.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿export class AppVeyorServiceToken {
+  static token = 'i2l609q1sfes3m24pwgf';
```
Comment (felix-b) [inline Source/Presentation/Web/AppVeyorPoc/src/app/app-veyor-service-token.ts]: @ILyaIL83, did you intend to commit this token?

Comment (ILyaIL83) [inline Source/Presentation/Web/AppVeyorPoc/src/app/app-veyor-service-token.ts]: F@#k. No, it's bug that came with fix.(I've already regenerated my token).

**Comment by felix-b** [comment]
Maybe I forgot to mention it in earlier discussions - testing generated code is actually the way to test the generators. Whether those unit or e2e tests, or a combination of the two, can be decided. 

Also, there will be static components, services, modules, etc (infrastructure layer) - which are manually coded and not generated. Those need to be tested as well. 

We have to learn how we include web unit/e2e tests in our development workflow, including the CI of course. This can be part of current POC or a separate one.

**Comment by ILyaIL83** [comment]
Site configuration moved to AppVeyorPoc\configuration.ts:
`export class Configuration {
  static appVeyorToken = 'your_token';
}`
but excluded from version controlling(should be added manually), while all site(src/**) forcefully included.

**Comment by ILyaIL83** [comment]
Please give your thoughts about 
- unit testing
- integration tests
- CI
regarding web client here https://github.com/felix-b/NWheels/issues/44

**Comment by felix-b** [comment]
Now it works! Super.

**Comment by ILyaIL83** [comment]
Review response:
1. Merge done(all checks have passed)
2. Have no idea what are you talking about yet.
3. For tests I've created issue, so yes we have another POC https://github.com/felix-b/NWheels/issues/51
4. Hope, after my responses, it's all clear.

**Comment by ILyaIL83** [comment]
angular2-webpack-starter removed, commit done.

**Pull request merged**